includes/class-fs-garbage-collector.php

```diff
@@ -0,0 +1,318 @@
+
+                foreach ( $all_product_data as $slug => $product_data ) {
+                    if ( isset( $products_to_skip_by_type_and_slug[ $product_type ][ $slug ] ) ) {
+                        continue;
+                    }
+
+                    if ( ! is_object( $product_data ) ) {
+                        continue;
+                    }
+
+                    if (
+                        empty( $product_data->last_load_timestamp ) ||
+                        ! is_numeric( $product_data->last_load_timestamp )
+                    ) {
+                        continue;
+                    }
+
+                    if ( $product_data->last_load_timestamp > ( time() - ( WP_FS__TIME_WEEK_IN_SEC * 4 ) ) ) {
+                        // Do not remove the data if the last activation was within the last 4 weeks.
+                        continue;
+                    }
+
+                    $is_addon = FS_Plugin::is_valid_id( $product_data->parent_plugin_id );
+
+                    if ( ! isset( $products_slugs_by_type[ $product_type ] ) ) {
+                        $products_slugs_by_type[ $product_type ] = array();
+                    }
+
+                    $product = array(
+                        'id'   => $product_data->id,
+                        'slug' => $slug,
+                        'file' => $product_data->file,
+                    );
+
+                    if ( ! $is_addon ) {
+                        $products_slugs_by_type[ $product_type ][] = $product;
+                    } else {
+                        /**
+                         * If add-on, add to the beginning of the array so that add-ons are removed before their parent. This is to prevent an unexpected issue when an add-on exists but its parent was already removed.
+                         */
+                        array_unshift( $products_slugs_by_type[ $product_type ], $product );
+                    }
+
+                    $products_data_by_type[ $product_type ][ $option_name ] = $product_data;
+                }
+            }
+
+            if ( empty( $products_slugs_by_type ) ) {
+                return;
+            }
+
+            $products_options_by_type = $this->get_product_options_by_type();
+
+            $loaded_products_options_by_type = array_fill_keys( $product_types, false );
+
+            $installs_count_by_user_id = array();
+
+            $has_updated_option = false;
+
+            foreach( $products_slugs_by_type as $product_type => $products ) {
+                foreach( $products as $product ) {
+                    $product_id   = $product['id'];
+                    $slug         = $product['slug'];
+                    $product_file = $product['file'];
+
+                    if ( ! $loaded_products_options_by_type[ $product_type ] ) {
+                        $products_data_by_type[ $product_type ]           = $this->load_options( $products_options_by_type[ $product_type ] );

```

Comment:
@fajardoleo aren't you overriding what was set on line `116`. Do we really need both?